sherpa/astro/io/pyfits_backend.py

```diff
@@ -263,6 +264,66 @@ def _try_vec_or_key(hdu, name, size, dtype=SherpaFloat, fix_type=False):
 
 # Read Functions
 
+# Crates supports reading gzipped files both when the '.gz' suffix
+# is given and even when it is not (i.e. if you ask to load 'foo.fits'
+# and 'foo.fits.gz' exists, then it will use this). This requires some
+# effort to emulate with the astropy backend.
+
+def is_binary_file(fname):
+    """Do we think the file contains binary data?
+
+    Parameters
+    ----------
+    fname : str
+       The file name.
+
+    Returns
+    -------
+    flag : bool
+       ``True`` if the file appears to contain binary data,
+       ``False`` otherwise.
+
+    Notes
+    -----
+    This is a wrapper around the version in sherpa.utils which
+    attempts to transparently support reading from a `.gz`
+    version if the input file name does not exist.
+    """
+
+    if not os.path.exists(fname):
+        fname = fname + '.gz'
```

Comment:
Several issues here:

* I am not convinced `is_binary_file` is a good idea in the first place, but this is not the PR to work on this issue

* if the routine were to fail if both `fname` and `fname` + `.gz` do not exist (I think that's what you are suggesting) then we need to make sure the same error is raised as is done by `is_binary_file`: is it `IOError` or `IOErr` (hopefully no other choice). This would then lead to a coupling of concerns - e.g. if we change `is_binary_file` to raise a different error then it would also have to be changed here.

* We should probably have a discussion at some point about Sherpa's error class (I believe I was one of the people originally wishing for a sherpa-related error class, so I shouldn't complain, but I'm not sure the pain is worth the gain). If we got rid of the possibility of `IOErr` then it's more obvious that raising `IOError` here is an okay thing to do. Actually, I think this is likely to be part of our Python 3 transition discussion, so should probably be thought about then.

The item is the issue here, so it's a discussion between code clarity and having repeated code.